cockatrice/src/spoilerbackgroundupdater.h

```diff
@@ -0,0 +1,41 @@
+#ifndef COCKATRICE_TIMERTHREAD_H
+#define COCKATRICE_TIMERTHREAD_H
+
+#include <QObject>
+#include <QTimer>
+#include <QNetworkAccessManager>
+
+#define GITHUB_SPOILERS_STATUS_URL "https://raw.githubusercontent.com/Cockatrice/Magic-Spoiler/files/spoilerStatus.txt"
+#define GITHUB_SPOILERS_DOWNLOAD_URL "https://raw.githubusercontent.com/Cockatrice/Magic-Spoiler/files/spoiler.xml"
+
+class SpoilerBackgroundUpdater : public QObject
+{
+    Q_OBJECT
+public:
+    SpoilerBackgroundUpdater(QObject *apParent = nullptr);
+    ~SpoilerBackgroundUpdater();
+
+public slots:
+    void timeoutOccurredTimeToDownloadSpoilers();
+    void handleNewTimeInterval();
+
+private slots:
+    void actDownloadFinishedSpoilersFile();
+    void actSeeIfSpoilerSeasonIsActive();
+    void changeActiveStatus();
+
+private:
+    QTimer *mpTimerForSpoilers;
+    QNetworkAccessManager *mpNetworkAccessManager;
```

Comment:
When you add a pointer member to a class, always ensure you initialize it in the class constructor, either instnciating the object or setting it to nullptr